container-engine-lib/lib/backend_impls/docker/docker_manager/docker_manager.go

```diff
@@ -1245,6 +1274,111 @@ func (manager *DockerManager) FetchLatestImage(ctx context.Context, dockerImage
 	return nil
 }
 
+func (manager *DockerManager) BuildImage(ctx context.Context, imageName string, imageBuildSpec *image_build_spec.ImageBuildSpec) error {
+	contextDirPath := imageBuildSpec.GetContextDir()
+	containerImageFileTarReader, err := getBuildContext(contextDirPath)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred retrieving the build context for '%v' at context directory path: %v", imageName, contextDirPath)
+	}
+
+	// This whole sessions thing was blocking us from using Docker build, until we found this:
+	// https://github.com/hashicorp/waypoint/pull/1937
+	uuidStr, err := uuid_generator.GenerateUUIDString()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred generating a UUID to give the Docker Buildkit session")
+	}
+	sessionName := fmt.Sprintf("kurtosis-%s", uuidStr)
+
+	// We generate a new session every time because, per https://github.com/moby/buildkit/issues/1432 ,
+	// sharing sessions is an optimization
+	// We don't bother reusing sessions so that we don't hit bugs
+	buildkitSession, err := session.NewSession(ctx, sessionName, buildkitSessionSharedKey)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error")
+	}
+
+	dialSessionFunc := func(ctx context.Context, proto string, meta map[string][]string) (net.Conn, error) {
```

Comment:
replace this with the `bkclient.New` function that makes this call under the hood - more explicit that what's actually happening is a client to buildkit is being established

Reply:
https://github.com/moby/moby/blob/311b9ff0aa93aa55880e1e5f8871c4fb69583426/client/buildkit/buildkit.go#L16